--- numpy-1.3.0/numpy/core/include/numpy/npy_cpu.h
+++ numpy-1.3.0/numpy/core/include/numpy/npy_cpu.h	2009-04-05 10:09:20.000000000 +0200
@@ -9,6 +9,7 @@
  *              NPY_CPU_S390
  *              NPY_CPU_IA64
  *              NPY_CPU_PARISC
+ * 		 NPY_CPU_ARM
  */
 #ifndef _NPY_CPUARCH_H_
 #define _NPY_CPUARCH_H_
@@ -45,6 +46,12 @@
 #elif defined(__parisc__)
     /* XXX: Not sure about this one... */
     #define NPY_CPU_PARISC
+#elif defined(__ARM__) || defined(__arm__)
+#	if defined(__ARMEB__)
+		#define NPY_CPU_ARMEB
+#	else
+		#define NPY_CPU_ARMEL
+#	endif
 #else
     #error Unknown CPU, please report this to numpy maintainers with \
     information about your platform (OS, CPU and compiler)
---numpy-1.3.0/numpy/core/include/numpy/npy_endian.h
+++ numpy-1.3.0.backup/numpy/core/include/numpy/npy_endian.h	2009-04-05 10:09:20.000000000 +0200
@@ -22,11 +22,12 @@
     #include "npy_cpu.h"
 
     #if defined(NPY_CPU_X86) || defined(NPY_CPU_AMD64)\
-            || defined(NPY_CPU_IA64)
+            || defined(NPY_CPU_IA64) || defined(NPY_CPU_ARMEL)
         #define NPY_LITTLE_ENDIAN
         #define NPY_BYTE_ORDER 1234
     #elif defined(NPY_CPU_PPC) || defined(NPY_CPU_SPARC)\
-            || defined(NPY_CPU_S390) || defined(NPY_CPU_PARISC)
+            || defined(NPY_CPU_S390) || defined(NPY_CPU_PARISC)\
+	    || defined(NPY_CPU_ARMEB)
         #define NPY_BIG_ENDIAN
         #define NPY_BYTE_ORDER 4321
     #else
